### PR TITLE
Join operator API improvements.

### PR DESCRIPTION
**Pull request by dmvk**

### CODE REVIEW

**Comment by dmvk** [comment]
Related to #143

File: euphoria-core/src/main/java/cz/seznam/euphoria/core/client/operator/InnerJoin.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+import cz.seznam.euphoria.core.annotation.audience.Audience;
+import cz.seznam.euphoria.core.client.dataset.Dataset;
+import cz.seznam.euphoria.core.client.functional.BinaryFunctor;
+import cz.seznam.euphoria.core.client.functional.UnaryFunction;
+
+import java.util.Objects;
+
+@Audience(Audience.Type.CLIENT)
+public class InnerJoin {
```
Comment (je-ik) [inline euphoria-core/src/main/java/cz/seznam/euphoria/core/client/operator/InnerJoin.java]: This operator should remain just `Join`.

File: euphoria-core/src/main/java/cz/seznam/euphoria/core/client/operator/RightJoin.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+import cz.seznam.euphoria.core.annotation.audience.Audience;
+import cz.seznam.euphoria.core.client.dataset.Dataset;
+import cz.seznam.euphoria.core.client.functional.BinaryFunctor;
+import cz.seznam.euphoria.core.client.functional.UnaryFunction;
+
+import java.util.Objects;
+import java.util.Optional;
+
+@Audience(Audience.Type.CLIENT)
+public class RightJoin {
```
Comment (je-ik) [inline euphoria-core/src/main/java/cz/seznam/euphoria/core/client/operator/RightJoin.java]: How is this implementation different from `LeftJoin`? Isn't this just quite expensive syntactic sugar around the same implementation with swapped arguments?

File: euphoria-operator-testkit/src/main/java/cz/seznam/euphoria/operator/test/JoinTest.java [hunk 3]
```diff
@@ -101,6 +143,45 @@ public void batchJoinOuter() {
...
         return Arrays.asList(
             Pair.of(0, "0+null"),
+            Pair.of(2, "2+12"), Pair.of(2, "2+12"), Pair.of(4, "4+14"),
+            Pair.of(1, "1+11"), Pair.of(1, "1+11"),
+            Pair.of(3, "3+13"), Pair.of(3, "3+13"));
+      }
+    });
+  }
+
+  @Processing(Processing.Type.BOUNDED)
```
Comment (je-ik) [inline euphoria-operator-testkit/src/main/java/cz/seznam/euphoria/operator/test/JoinTest.java]: Why do we run these tests only for bounded datasets?

Comment (je-ik) [reply euphoria-operator-testkit/src/main/java/cz/seznam/euphoria/operator/test/JoinTest.java]: OK, I see, the reason is probably that unbounded dataset cannot be join using global windowing. There are separate tests for windowing.

**Pull request merged**